.github: Update PR template

andreittr opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!
We welcome new changes, features, fixes, and more!
Please fill in this short form to indicate the status of your PR.

-->

### Description of Changes

<!--
Provide a detailed description of the changes made in this PR.

This should include, as applicable:
- Context & motivation (e.g., fixes bug X, introduces feature that enables Y, etc.)
- Overview of code changes (what has been changed and to what end)
- Public interface changes (library API(s), syscall API/ABI, Kconfig, etc.)
- Any other notable mentions regarding review, testing, and integration
-->

This change updates the PR template to prioritize relevant information and streamline the PR process.

Notable changes:
- Moved "Description of changes" to the top, and added guiding points in the comment
- Removed "Additional configuration", folding it into a point in "Description of changes"
- Tweaked "Base target" to be more usable OOTB, but still allow an easy overview if a change impacts only a particular arch/plat/app, etc.; renamed to "Impact Scope"
- Moved PR checklist to last step, as a final check flowing naturally before pressing the "Create pull request" button

In the spirit of things, this present PR description is styled in the new template.

### Related Work

<!--
Link here any open PRs that this work depends on or which it will conflict with.
-->

N/A

### PR Checklist

<!--
Finally, review and mark prerequisites appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

andreittr committed:
.github: Update PR template

This change updates the PR template to prioritize relevant information
and streamline the PR process.

Signed-off-by: Andrei Tatar <ttr@unikraft.io>

nderjung commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -9,38 +9,38 @@ before opening a new PR as this will cover the PR process:
+
+ - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
+ - [ ] Tested your changes against relevant architectures and platforms;
+ - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
```
Check patch is now run in the CI - i think when this was originally created; it was not the case and run my a maintainer. So potentially, this line can now be removed? The alternative is of course to keep it so that people are aware (also fine). In the latter case, maybe in `<!--` a comment `-->` it could show how to run it?

andreittr replied:
> The alternative is of course to keep it so that people are aware (also fine).

Precisely that's why I still kept it, I use it myself to remind me to have run `checkpatch.uk` before pushing that green button :smile: 

> maybe in <!-- a comment --> it could show how to run it?

Good point, will add :+1:

michpappas reviewed (commented):
LGTM. Only one thing to consider is the use of "please", which is generally discouraged by various UX guidelines. Some examples are provided in: https://ux.stackexchange.com/questions/25500/why-avoid-the-word-please-in-error-messages. Also:
* https://developers.google.com/style/word-list#please
* https://developers.google.com/style/tone#politeness

Also I'm not sure whether the impact scope table has ever been useful, but I don't have much of a strong opinion about it.

andreittr commented:
Pushed update; specifically:
- removed "Impact Scope" section (we nearly never used its equivalent)
- removed all but the first instance of the word "please"; I decided to keep it _just_ for the one instance in which we ask something inconvenient -- filling in the form itself
- cleaned up introductory comment to be shorter and to the point; removed the contrib guidelines link, as it's already linked in the checklist (and probably other docs too)
- added "Related Work" section; from personal experience this is the one most added thing to our PRs besides the description itself

michpappas reviewed (approved):
Reviewed-by: Michalis Pappas <michalis@unikraft.io>

mogasergiu reviewed (approved):
Reviewed-by: Sergiu Moga <sergiu@unikraft.io>

michpappas commented:
Approved-by: Michalis Pappas <michalis@unikraft.io>
